tests/Unit/Evolution/Systems/RelativisticEuler/Valencia/Test_Characteristics.cpp

```diff
@@ -0,0 +1,92 @@
+#include "tests/Unit/Pypp/Pypp.hpp"
+#include "tests/Unit/Pypp/SetupLocalPythonEnvironment.hpp"
+#include "tests/Utilities/MakeWithRandomValues.hpp"
+
+// IWYU pragma: no_forward_declare Tensor
+
+namespace {
+
+template <size_t Dim>
+void test_characteristic_speeds(const DataVector& used_for_size) noexcept {
+  //  Arbitrary random numbers can produce a negative radicand in Lambda^\pm.
+  //  This bound helps to prevent that situation.
+  const double max_value = 1.0 / sqrt(Dim);
+  pypp::check_with_random_values<6>(
+      &RelativisticEuler::Valencia::characteristic_speeds<Dim>, "TestFunctions",
+      "characteristic_speeds",
+      {{{0.0, 1.0},
+        {-1.0, 1.0},
+        {-max_value, max_value},
+        {0.0, 1.0},
+        {0.1, 1.0},
+        {-max_value, max_value}}},
+      used_for_size);
+}
+
+template <size_t Dim>
+void test_with_normal_along_coordinate_axes(
+    const DataVector& used_for_size) noexcept {
+  const auto seed = std::random_device{}();
+  CAPTURE(seed);
+  std::mt19937 generator(seed);
+  std::uniform_real_distribution<> distribution(0.0, 1.0);
+
+  const auto nn_generator = make_not_null(&generator);
+  const auto nn_distribution = make_not_null(&distribution);
+
+  const auto lapse = make_with_random_values<Scalar<DataVector>>(
+      nn_generator, nn_distribution, used_for_size);
+  const auto shift = make_with_random_values<tnsr::I<DataVector, Dim>>(
+      nn_generator, nn_distribution, used_for_size);
+  const auto spatial_velocity =
+      make_with_random_values<tnsr::I<DataVector, Dim>>(
+          nn_generator, nn_distribution, used_for_size);
+  const auto spatial_velocity_squared =
+      make_with_random_values<Scalar<DataVector>>(nn_generator, nn_distribution,
+                                                  used_for_size);
+  const auto sound_speed_squared = make_with_random_values<Scalar<DataVector>>(
+      nn_generator, nn_distribution, used_for_size);
+
+  for (const auto& direction : Direction<Dim>::all_directions()) {
+    const auto normal = euclidean_basis_vector(direction, used_for_size);
+
+    CHECK_ITERABLE_APPROX(
+        RelativisticEuler::Valencia::characteristic_speeds(
+            lapse, shift, spatial_velocity, spatial_velocity_squared,
+            sound_speed_squared, normal),
+        (pypp::call<std::array<DataVector, Dim + 2>>(
+            "TestFunctions", "characteristic_speeds", lapse, shift,
+            spatial_velocity, spatial_velocity_squared, sound_speed_squared,
+            normal)));
+  }
+}
+
+}  // namespace
+
+SPECTRE_TEST_CASE("Unit.RelativisticEuler.Valencia.Characteristics",
+                  "[Unit][Evolution]") {
+  pypp::SetupLocalPythonEnvironment local_python_env{
+      "Evolution/Systems/RelativisticEuler/Valencia"};
+
+  GENERATE_UNINITIALIZED_DATAVECTOR;
+  CHECK_FOR_DATAVECTORS(test_characteristic_speeds, (1, 2, 3))
+  CHECK_FOR_DATAVECTORS(test_with_normal_along_coordinate_axes, (1, 2, 3))
```

Comment:
What is the reason for testing a specific case in addition to the general case? If there is something specific you are testing for, please add a comment.

Reply:
This was suggested for the Newtonian chars, where boundaries perpendicular to the coordinate axes are of particular interest. It might be unnecessary here, though.

Reply:
I like testing along the axes

Reply:
@kidder @carmaza @fmahebert would you like a comment or no? Ya'll get one vote :P I don't vote so we can't have draw :D

Reply:
I would like a comment, perhaps similar to "Test with aligned normals to check the code works with vector components being 0.". This would clear up the questions I had when originally reading the code.

Reply:
I was gonna say no unless @fmahebert finds it useful. Therefore, I added the comment.